extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/ResteasyReactiveServerRuntimeConfig.java

```diff
@@ -2,37 +2,34 @@
 
 import java.nio.charset.Charset;
 
-import io.quarkus.runtime.annotations.ConfigGroup;
-import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
 
-@ConfigRoot(name = "resteasy-reactive", phase = ConfigPhase.RUN_TIME)
-public class ResteasyReactiveServerRuntimeConfig {
+@ConfigMapping(prefix = "quarkus.resteasy-reactive")
+@ConfigRoot(phase = ConfigPhase.RUN_TIME)
+public interface ResteasyReactiveServerRuntimeConfig {
 
     /**
      * Input part configuration.
      */
-    @ConfigItem
-    public MultipartConfigGroup multipart;
+    MultipartConfigGroup multipart();
 
-    @ConfigGroup
-    public static class MultipartConfigGroup {
+    interface MultipartConfigGroup {
 
         /**
          * Input part configuration.
          */
-        @ConfigItem
-        public InputPartConfigGroup inputPart;
+        InputPartConfigGroup inputPart();
     }
 
-    @ConfigGroup
-    public static class InputPartConfigGroup {
+    interface InputPartConfigGroup {
 
         /**
          * Default charset.
          */
-        @ConfigItem(defaultValue = "UTF-8")
-        public Charset defaultCharset;
+        @WithDefault("UTF-8")
+        Charset defaultCharset();
```

Comment:
@radcortez in a lot of config groups, we have methods like `boolean isAnyPropertySet()`. By making the `@ConfigItem` element not mandatory (or even not desirable), my understanding is that we will need a way to mark these methods as ignored from the config (and the doc)?

Moving these methods outside of the interface will be counter productive as I'm pretty sure they will end up not being updated when we add new properties.

And I suppose we will have to make them default methods.

Reply:
Wouldn't simply making the method default be enough for Smallry Config to ignore them?>

Reply:
I have no idea thus why I'm asking. But they also need to be ignored in the doc. Not sure if @radcortez already took that into account.

Reply:
Yes, default methods are ignored :)

I fixed that recently here: https://github.com/quarkusio/quarkus/pull/31384